File: whenylibrary/src/main/java/com/example/commonlibrary/annotation_api/ViewModelInjectorApi.java
package com.example.commonlibrary.annotation_api;

import android.app.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.ViewDataBinding;
import androidx.lifecycle.ViewModelProvider;

public interface ViewModelInjectorApi {

    <T extends ViewDataBinding> T injectViewModels(AppCompatActivity activity, int layoutId);

    void injectViewModelByFactory(AppCompatActivity activity, ViewModelProvider.NewInstanceFactory factory, ViewDataBinding viewDataBinding);

}

File: wheny_annotation_apilib/src/main/java/com/example/whenyannotationapilib/ViewModelInjectorApi.kt
package com.example.whenyannotationapilib

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.ViewDataBinding
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory

interface ViewModelInjectorApi {

    fun <T : ViewDataBinding?> injectViewModels(tag: Any?, activity: AppCompatActivity?, inflater: LayoutInflater?, layoutId: Int, viewGroup: ViewGroup?): T?{
        return null
    }


    fun injectViewModelByFactory(tag: Any?, activity: AppCompatActivity?, factory: NewInstanceFactory?, filedName: String?, viewDataBinding: ViewDataBinding?){


    }

}